src/protocols/hpr_gwmp_router.erl

```diff
@@ -0,0 +1,112 @@
+-module(hpr_gwmp_router).
+
+-include("../grpc/autogen/server/packet_router_pb.hrl").
+
+-export([send/3]).
+
+-export([
+    packet_up_to_push_data/2,
+    route_to_dest/1,
+    txpk_to_packet_down/1
+]).
+
+-spec send(
+    Packet :: hpr_packet_up:packet(),
+    Stream :: grpcbox_stream:t(),
+    Routes :: hpr_route:route()
+) -> ok | {error, any()}.
+send(PacketUp, Stream, Route) ->
+    Hotspot = hpr_packet_up:hotspot(PacketUp),
+
+    case hpr_gwmp_udp_sup:maybe_start_worker(Hotspot, #{socket_dest => route_to_dest(Route)}) of
```

Comment:
Should `route_to_dest/1`  live in the udp_worker?
The worker knows what it needs to start.
And moving the args creation and use closer together is 1 less thing that could become mismatched in the future.

Reply:
It doesn't feel right to have 
```    Lns = hpr_route:lns(Route),
```
in the worker. The udp worker shouldn't know about routes.

I'm not sure if this is an improvement, but you could pass the 'Lns' above to the worker, and then the worker can process it into the desired form of {Address, Port}.

Reply:
That's a good point. 
Let's leave it where it is right now, and hopefully a better place will reveal itself if there is one.